lib/team_api/joiner.rb

```diff
@@ -72,8 +71,8 @@ def team_member_key(ref)
     end
 
     def team_member_from_reference(reference)
-      key = team_member_key reference
-      team[key] || team[team_by_email[key] || team_by_github[key.downcase]]
+      key = (team_member_key reference).downcase
```

Comment:
Nope, I agree - this is a case of me waffling on Ruby idioms vs. code readability.  I should trust my gut, go with readability. :-)